detection-rules/attachment_powershell_content.yml

```
@@ -0,0 +1,26 @@
+name: "Attachment: PowerShell Content"
+description: |
+  Recursively scans files and archives to detect PowerShell content.
+
+  While scripts are often blocked by mail filtering, alternative file formats and archived content may be employed to bypass such controls.
+references:
+  - https://stackoverflow.com/questions/62604621/what-are-the-different-powershell-file-types
+  - https://en.wikipedia.org/wiki/PowerShell#:~:text=named%20native%20commands.-,Filename%20extensions,-%5Bedit%5D
+type: "rule"
+authors:
+  - twitter: "ajpc500"
+source: |
+  type.inbound and
+  any(attachments,
+      .file_extension in~ (
+          // PowerShell related file extensions
+          "ps1", "ps1xml", "psm1", "psd1", "pssc", "psrc", "cdxml", "ps2", "ps2xml", "psc2",
+          // Archive file formats 
+          "7z","bz2","gz","rar","tar","zip","zipx","iso","img"
+      )
+      and any(beta.binexplode(.),.extension in 
```

```suggestion
      and any(beta.binexplode(.), .extension in~
```

we might want to make this case insensitive too

---

Ah yeah, good spot 👍🏻